torch/distributed/_tensor/ops/tensor_ops.py

```diff
@@ -64,6 +64,11 @@ def default_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> StrategyType:
         aten.empty_like.default,
         aten.full_like.default,
         aten.ones_like.default,
+        aten.rand_like.default,
+        aten.randn_like.default,
+        aten.randint_like.default,
+        aten.randint_like.low_dtype,
+        aten.randint_like.low_dtype_out,
```

Comment:
I think we should add RuntimeSchemaInfo that includes "dtype" kwarg so that sharding propagation cache different dtype configs correctly. 

like this https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/ops/tensor_ops.py#L109